Makefile

```diff
@@ -55,3 +63,28 @@ coverage:
 coveralls: coverage
 	go get github.com/mattn/goveralls
 	goveralls -coverprofile=$(COVERAGE_FILE) -service=github
+
+.PHONY: bench-deps
+bench-deps:
+	rm -rf ${BENCH_PATH}
+	mkdir ${BENCH_PATH}
+	go clean -testcache
+	cd ${BENCH_PATH} && git clone https://go.googlesource.com/perf && cd perf && go install ./cmd/benchstat
+	rm -rf ${BENCH_PATH}/perf
+
+.PHONY: bench
+bench:
+	go test ${TEST_PATH} -bench=. -run=^Benchmark -benchmem -benchtime=${DURATION} -count=${COUNT} 2>&1 \
+		| grep -v -E 'pkg:|ok|\?|PASS' | tee ${BENCH_FILE}
+	benchstat ${BENCH_FILE}
+
+.PHONY: bench-reference
+bench-reference:
```

Comment:
I think it is worth to add a message for each bench-* target with a short description. (like we did for target with functional tests)

Reply:
Added.